base/pkg/libgit2/config.jl

```diff
@@ -0,0 +1,78 @@
+# This file is a part of Julia. License is MIT: http://julialang.org/license
+
+function GitConfig(path::AbstractString)
+    cfg_ptr_ptr = Ref{Ptr{Void}}(C_NULL)
+    err = ccall((:git_config_open_ondisk, :libgit2), Cint,
+                 (Ptr{Ptr{Void}}, Ptr{UInt8}), cfg_ptr_ptr, path)
+    err !=0 && return nothing
+    return GitConfig(cfg_ptr_ptr[])
+end
+
+function GitConfig(r::GitRepo)
+    cfg_ptr_ptr = Ref{Ptr{Void}}(C_NULL)
+    err = ccall((:git_repository_config, :libgit2), Cint,
+                 (Ptr{Ptr{Void}}, Ptr{Void}), cfg_ptr_ptr, r.ptr)
+    err !=0 && return nothing
+    return GitConfig(cfg_ptr_ptr[])
+end
+
+function GitConfig()
+    cfg_ptr_ptr = Ref{Ptr{Void}}(C_NULL)
+    ccall((:git_config_open_default, :libgit2), Cint,
+          (Ptr{Ptr{Void}}, ), cfg_ptr_ptr)
+    return GitConfig(cfg_ptr_ptr[])
+end
+
+function get{T}(::Type{T}, c::GitConfig, name::AbstractString)
```

Comment:
These ifelse clauses are very non-Julian.  It would be better to have separate `get{T<:AbstractString}(::Type{T}, ...)`, `get(::Type{Bool}, ...)` etcetera methods.